cmd/root.go

```diff
@@ -45,6 +46,17 @@ func NewRootCmd(out io.Writer, args []string) *cobra.Command {
 	flags.Parse(args)
 	settings = new(EnvSettings)
 
+	// Helm plugins are not passed Helm's global flags but instead must
+	// read the global flag values from each corresponding environment variable.
```

Comment:
Might read better as:
" Helm plugins are not passed the plugin flags when run with the Helm plugin framework e.g. helm 2to3 convert ...
  The flag values are set to corresponding environment variables instead. The flags are passed as expected when run directly using the binary."

Reply:
How about modifying yours just a bit to clarify that it is Helm's global flags that are not passed to the plugins?  Like so:

"When run with the Helm plugin framework, Helm plugins are not passed the plugin flags that correspond to Helm global flags e.g. helm 2to3 convert --kube-context ...
The flag values are set to corresponding environment variables instead. The flags are passed as expected when run directly using the binary."

Reply:
I think thats good.